### PR TITLE
feat: optimize vacuum drop table, execute in parallel

### PR DESCRIPTION
**Pull request by lichuang**
I hereby agree to the terms of the CLA available at: https://docs.databend.com/dev/policies/cla/

## Summary

feat: optimize vacuum drop table, execute in parallel

- Fixes #[Link the issue here]

## Tests

- [x] Unit Test
- [ ] Logic Test
- [ ] Benchmark Test
- [ ] No Test  - _Explain why_

## Type of change

- [ ] Bug Fix (non-breaking change which fixes an issue)
- [ ] New Feature (non-breaking change which adds functionality)
- [ ] Breaking Change (fix or feature that could cause existing functionality not to work as expected)
- [ ] Documentation Update
- [ ] Refactoring
- [x] Performance Improvement
- [ ] Other (please describe):

- - -
This change is [Reviewable](https://reviewable.io/reviews/datafuselabs/databend/15478)

### CODE REVIEW

File: src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs [hunk 1]
```diff
@@ -15,120 +15,143 @@
...
+            threads_nums * 2,
+            "batch-vacuum-drop-tables-worker".to_owned(),
+        )
+        .await?;
+        if dry_run_limit.is_some() {
+            let mut ret_files = vec![];
+            for file in result {
+                if let Ok(Some(files)) = file {
+                    ret_files.extend(files);
+                }
```
Comment (dantengsky) [inline src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs:139]: This ignores any exceptions that might occur during the executions of `do_vacuum_drop_table`, which is inconsistent with the previous behavior, should we consider throwing the exceptions?

**Comment by dantengsky** [comment]
besides:

- cargo clippy complaints
   https://github.com/datafuselabs/databend/actions/runs/9041307782/job/24846563784?pr=15478#step:4:4786
   
- with this PR, the meaning of the parameter `dry_runt_limit` will change

the number of files returned might be slightly larger than dry_runt_limit. we either need to update the documentation accordingly or adjust the list of files that are returned (at the end of execution).

File: src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs (hunk 1 above)
Comment (lichuang) [reply src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs]:
```                
// return error if any errors happens during `do_vacuum_drop_table`
                if let Err(e) = file {
                    return Err(e);
                } else if let Ok(Some(files)) = file {
                    ret_files.extend(files);
                }
            }
```

File: src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs [hunk 2]
```diff
@@ -28,108 +29,133 @@ use opendal::Operator;
...
+        .await?;
+        if dry_run_limit.is_some() {
+            let mut ret_files = vec![];
+            for file in result {
+                // return error if any errors happens during `do_vacuum_drop_table`
+                if let Err(e) = file {
+                    return Err(e);
+                } else if let Ok(Some(files)) = file {
+                    ret_files.extend(files);
+                }
```
Comment (dantengsky) [inline src/query/ee/src/storages/fuse/operations/vacuum_drop_tables.rs:146]:
```suggestion
                if let Some(files) = file? {
                    ret_files.extend(files);
                }
```

**Review by dantengsky — APPROVED** [review]

**Pull request merged**